include/tc/core/cpu/cpu_mapping_options-inl.h

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "tc/core/cpu/cpu_mapping_options.h"
+
+namespace tc {
+
+CpuMappingOptions::CpuMappingOptions()
+    : ownedProto_(), genericMappingOptionsView(ownedProto_) {}
+
+/// Construct from a serialized protocol buffer message.
+CpuMappingOptions::CpuMappingOptions(const std::string& str)
+    : CpuMappingOptions() {
+  bool parsed = ownedProto_.ParseFromString(str);
+  CHECK(parsed) << "could not parse protobuf string";
```

Comment:
on a separate note, in Caffe2 earlier days we had to cut all the CHECKs out. Because they can crash a process and are not compatible with production workloads. So we use CAFFE_ENFORCE and CAFFE_THROW which use exceptions instead.

Reply:
yep, help on moving away from these cheks is much appreciated!  I was considering to define my own macros and just search/replace in the codebase, but if there is a better way -- let's do it.

Reply:
It may be easier to redefine CHECK.  See also #193.

Reply:
I would not redefine check, actually. As it is very magical. In Caffe2 we just did a codemod and replaced all CHECKs with CAFFE_ENFORCE / CAFFE_THROW

Also this gives ability to add custom exception being thrown. Which could have extra information.